ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -0,0 +1,872 @@
+ * by the caller. These data can be a object handle, a ciphered stream, etc.
+ *
+ * - Parameter #3 is not used.
+ */
+
+/*
+ * SKS_CMD_PING		Acknowledge TA presence and return TA versioning info
+ *
+ * param#0: none
+ * param#1: none
+ * param#2: none | out-memref : [uint32_t version1]
+ *				[uint32_t version2]
+ * param#3: none
+ */
+#define SKS_CMD_PING			0x00000000
+
+/*
+ * SKS_CMD_CK_SLOT_LIST - Get the table of the valid slot IDs
+ *
+ * param#0: none
+ * param#1: none
+ * param#2: out-memref : [uint32_t slot_ids]
+ * param#3: none
+ *
+ * The TA instance may represent several PKCS#11 slots and associated tokens.
+ * This command relates the PKCS#11 API function C_GetSlotList and return the
+ * valid IDs recognized by the trsuted application.
+ */
+#define SKS_CMD_CK_SLOT_LIST		0x00000001
+
+/*
+ * SKS_CMD_CK_SLOT_INFO - Get cryptoki structured slot information
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: out-memref : [struct sks_ck_slot_info info]
+ * param#3: none
+ *
+ * The TA instance may represent several PKCS#11 slots and associated tokens.
+ * This command relates the PKCS#11 API function C_GetSlotInfo and return the
+ * information abut the target slot.
+ */
+#define SKS_CMD_CK_SLOT_INFO		0x00000002
+
+struct sks_ck_slot_info {
+	uint8_t slotDescription[64];
+	uint8_t manufacturerID[32];
+	uint32_t flags;
+	uint8_t hardwareVersion[2];
+	uint8_t firmwareVersion[2];
+};
+
+/* Slot flags reflecting the pkcs11 flags */
+#define SKS_TOKEN_PRESENT		BIT(0UL)
+#define SKS_TOKEN_REMOVABLE		BIT(1UL)
+#define SKS_TOKEN_HW			BIT(2UL)
+
+/*
+ * SKS_CMD_CK_TOKEN_INFO - Get cryptoki structured token information
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: out-memref : [struct sks_ck_token_info info]
+ * param#3: none
+ *
+ * The TA instance may represent several PKCS#11 slots and associated tokens.
+ * This command relates the PKCS#11 API function C_GetTokenInfo and return the
+ * information abut the target represented token.
+ */
+#define SKS_CMD_CK_TOKEN_INFO		0x00000003
+
+#define SKS_TOKEN_LABEL_SIZE		32
+#define SKS_TOKEN_MANUFACTURER_SIZE	32
+#define SKS_TOKEN_MODEL_SIZE		16
+#define SKS_TOKEN_SERIALNUM_SIZE	16
+
+struct sks_ck_token_info {
+	uint8_t label[SKS_TOKEN_LABEL_SIZE];
+	uint8_t manufacturerID[SKS_TOKEN_MANUFACTURER_SIZE];
+	uint8_t model[SKS_TOKEN_MODEL_SIZE];
+	uint8_t serialNumber[SKS_TOKEN_SERIALNUM_SIZE];
+	uint32_t flags;
+	uint32_t ulMaxSessionCount;
+	uint32_t ulSessionCount;
+	uint32_t ulMaxRwSessionCount;
+	uint32_t ulRwSessionCount;
+	uint32_t ulMaxPinLen;
+	uint32_t ulMinPinLen;
+	uint32_t ulTotalPublicMemory;
+	uint32_t ulFreePublicMemory;
+	uint32_t ulTotalPrivateMemory;
+	uint32_t ulFreePrivateMemory;
+	uint8_t hardwareVersion[2];
+	uint8_t firmwareVersion[2];
+	uint8_t utcTime[16];
+};
+
+/* Flags (reflect the cryptoki flags) */
+#define SKS_TOKEN_HAS_RNG		BIT(0UL)	/* CKF_RNG */
```

Comment:
Why not follow the cryptoki names more closely? Random renaming doesn't help follow the mappings or grep the source code.
Also, having a common prefix in the struct and the defines would be nicer IMO.
```C
struct sks_token_info {
	...
};

/* Values for sks_token_info::flags. SKS_TOKEN_<x> corresponds to cryptoki flag CKF_<x> */
#define SKS_TOKEN_WRITE_PROTECTED	BIT(1)
#define SKS_TOKEN_LOGIN_REQUIRED		BIT(2)
...
```

Reply:
Ok for the aligment of labels with the pkcs11. I will fix that.